tests/bud.bats

```diff
@@ -1640,13 +1641,42 @@ function _test_http() {
   run_buildah from ${target}
 }
 
-@test "bud-github-context-with-branch-and-subdir" {
+# Containerfile in this repo should only exists on older commit and
+# not on HEAD or the default branch.
+@test "bud-github-context-from-commit" {
+  if ! which git ; then
+    skip "no git in PATH"
+  fi
+  target=giturl-image
+  # Any repo would do, but this one is small, is FROM: scratch, local, and has
+  # its entire build context in a subdirectory of the repository.
+  if ! start_git_daemon ${TEST_SOURCES}/git-daemon/repo-with-containerfile-on-old-commit.tar.gz ; then
+    skip "error running git daemon"
+  fi
+  # Containerfile in this repo should only exists on older commit and
+  # not on HEAD or the default branch.
+  gitrepo=git://localhost:${GITPORT}/repo#f94193d34548eb58650a10a5183936d32c2d3280
+  tmpdir="${TEST_SCRATCH_DIR}/build"
+  mkdir -p "${tmpdir}"
+  TMPDIR="${tmpdir}" run_buildah build $WITH_POLICY_JSON -t ${target} "${gitrepo}"
+  # Verify that build must fail on default `main` branch since we
+  # don't have a `Containerfile` on main branch.
+  gitrepo=git://localhost:${GITPORT}/repo#main
+  TMPDIR="${tmpdir}" run_buildah 125 build $WITH_POLICY_JSON -t ${target} "${gitrepo}"
```

Comment:
I don't see the reason for using a different $TMPDIR for these two build attempts.  #4159 did it so that it could check that the directory the repository would be cloned into was empty when the build finished, but we're not doing that here.